python-package/lightgbm/basic.py

```diff
@@ -725,8 +725,40 @@ def _free_handle(self):
         if self.handle is not None:
             _safe_call(_LIB.LGBM_DatasetFree(self.handle))
             self.handle = None
+        self.need_slice = True
+        if self.used_indices is not None:
+            self.data = None
```

Comment:
Can you please explain this `if` check?

Reply:
This is to restore the state before `construct()`. When it is subset, the data should reset to None.

Reply:
And later it will be set in `get_data()`, right?
```
self.data = self.reference.data 
```

Reply:
Yeah